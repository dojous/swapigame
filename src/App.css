.App {
  text-align: center;
  
  background-color: #282c34;
  background-color: white;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  /* color: white; */



}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

div.loader {
  font-size: 200px;
}

.flex-container {
  padding: 0;
  margin: 0;
  list-style: none;
  display: flex;
  align-items: center;
  justify-content: center;
}
row {
  width: 100%;
  
}

.flex-item {
  padding: 50px;
}



/* .flex-item {
  background: tomato;
  padding: 5px;
  width: 200px;
  height: 150px;
  margin: 10px;
  line-height: 150px;
  color: white;
  font-weight: bold;
  font-size: 3em;
  text-align: center;
} */