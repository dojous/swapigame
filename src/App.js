import React, { lazy, Suspense, useEffect, useState } from "react";
import { Route, Redirect, Switch } from "react-router-dom";

import "./App.css";

import Home from "./components/home";
import OnePlayer from "./components/oneplayer";
import TwoPlayer from "./components/twoplayers";
import axios from "axios";
import { API_URL } from "./config";

import Loader from "./components/spinner";

function App() {
  const [resources, setRecources] = useState([]);

  const getRecources = async () => {
    await axios
      .get(`${API_URL}/`)
      .then(res => {
        setRecources(Object.keys(res.data));
      })

      .catch(err => {
        console.log("ERR", err);
      });
  };

  useEffect(() => {
    getRecources();
  }, []);

  if (resources.length === 0) {
    return (
      <div className="App">
        <Loader color="black" />
        <h1>Loading swapi-game data</h1>
      </div>
    );
  }

  return (
    <div className="App">
          <Switch>
        <Route path="/" exact render={props => <Home />} />
        <Route
          path="/oneplayer"
          render={props => <OnePlayer resources={resources} {...props} />}
        />

        <Route
          path="/twoplayer"
          render={props => <TwoPlayer resources={resources} {...props} />}
        />

        <Suspense
          fallback={
            <>
              <Loader color="black" />
              <h1>Loading page details</h1>
            </>
          }
        >
          <Route
            path={`/${"(" + resources.join("|") + ")"}/:id`}
            component={LazyItemDetaileds}
          />
        </Suspense>

        <Route path="/not-found" component={() => <div>Page not found</div>} />

        <Redirect to="/not-found" />
      </Switch>
    </div>
  );
}

const LazyItemDetaileds = () => {
  const ItemDetailed = lazy(() =>
    new Promise(resolve => setTimeout(resolve, 1)).then(resolve =>
      import("./components/itemdetailed")
    )
  );

  return <ItemDetailed />;
};

export default App;
