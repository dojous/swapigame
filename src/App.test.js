import React from "react";
import { render, act, waitForElementToBeRemoved } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";
import axiosMock from "axios";
import App from "./App";

jest.mock("axios");

describe("one player version", () => {
  test("should render correct list of resources", async () => {
    axiosMock.get.mockResolvedValueOnce({
      data: {
        people: "https://swapi.co/api/people/",
        planets: "https://swapi.co/api/planets/",
        films: "https://swapi.co/api/films/",
        species: "https://swapi.co/api/species/",
        vehicles: "https://swapi.co/api/vehicles/",
        starships: "https://swapi.co/api/starships/"
      }
    });
    await act(async () => {
      const { container, getByText } = render(
        <Router>
          <App />
        </Router>
      );
      await waitForElementToBeRemoved(() =>
        getByText("Loading swapi-game data")
      );
      expect(container).toMatchSnapshot("start screen");
    });
  });
});
