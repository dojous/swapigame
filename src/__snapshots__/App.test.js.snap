// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`one player version should render correct list of resources: start screen 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="flex-container"
    >
      <div
        class="row"
      >
        <div
          class="flex-item"
        >
           
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-1 MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              1 player
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="flex-item"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-1 MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              2 players -
               
              <p
                style="color: red; font-weight: 900;"
              >
                BONUS!
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
           
        </div>
      </div>
    </div>
  </div>
</div>
`;
